### PR TITLE
feat: add faucet page

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/rnd-dev-team#1548

- add faucet page for redwood testnet

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1. https://deploy-preview-2092--regen-marketplace.netlify.app/faucet

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by ryanchristo** [comment]
HTTPS and CORS should now be configured on our validator node server: https://redwood.regen.network/faucet/

File: web-marketplace/src/pages/Faucet/hook/useRequestFunds.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+export const useRequestFunds = () => {
+  const setErrorBannerTextAtom = useSetAtom(errorBannerTextAtom);
+  const setBannerTextAtom = useSetAtom(bannerTextAtom);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const requestFunds = async (address: string) => {
+    setIsLoading(true);
+    try {
+      const response = await fetch('https://redwood.regen.network:8000', {
```
Comment (ryanchristo) [inline web-marketplace/src/pages/Faucet/hook/useRequestFunds.tsx:15]:
```suggestion
      const response = await fetch('https://redwood.regen.network/faucet', {
```

**Review by ryanchristo — COMMENTED** [review]
Looks good to me. Will test once the url is updated.

**Comment by flagrede** [comment]
@ryanchristo I have updated the faucet endpoint but the call is still not working. I don't have a specific error this time, the call just timeout at some point. Might be because of the proxy?

File: web-marketplace/src/pages/Faucet/hook/useRequestFunds.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+export const useRequestFunds = () => {
+  const setErrorBannerTextAtom = useSetAtom(errorBannerTextAtom);
+  const setBannerTextAtom = useSetAtom(bannerTextAtom);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const requestFunds = async (address: string) => {
+    setIsLoading(true);
+    try {
+      const response = await fetch(
+        'https://redwood.regen.network:8000/faucet',
```
Comment (ryanchristo) [inline web-marketplace/src/pages/Faucet/hook/useRequestFunds.tsx:16]:
```suggestion
        'https://redwood.regen.network:8000/faucet/',
```

**Comment by ryanchristo** [comment]
Feel free to rebase, the only thing I noticed different with my setup was the trailing `/`, so wanted to quickly test.

File: web-marketplace/src/pages/Faucet/hook/useRequestFunds.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+export const useRequestFunds = () => {
+  const setErrorBannerTextAtom = useSetAtom(errorBannerTextAtom);
+  const setBannerTextAtom = useSetAtom(bannerTextAtom);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const requestFunds = async (address: string) => {
+    setIsLoading(true);
+    try {
+      const response = await fetch(
+        'https://redwood.regen.network:8000/faucet/',
```
Comment (ryanchristo) [inline web-marketplace/src/pages/Faucet/hook/useRequestFunds.tsx:16]:
```suggestion
        'https://redwood.regen.network/faucet/',
```

**Comment by ryanchristo** [comment]
And I just noticed you had the port still included.

**Review by ryanchristo — APPROVED** [review]
tACK. This seems to be working now.

**Comment by flagrede** [comment]
@ryanchristo Sorry for the port, missed that in the URL change.
I'm still having issues with the call, is it working on your hand?
image

**Comment by ryanchristo** [comment]
It's working for me. Looks like it might have been the trailing slash? 

Screenshot from 2023-09-18 10-24-45

**Pull request merged**